### PR TITLE
Toggle video playback when OK button is pressed

### LINKED ISSUE #883: OK button to pause
Press the OK button on remote during playback to pause. 

**Additional context**
I use many streaming services on Roku and all of them have this in common. During playback, a single press of OK will either pause the content, or show the overlay and with a second press will pause. 

Matching this in Jellyfin client would bring consistency to numerous other streaming services (Netflix, Hulu, Discovery...). This is specifically useful for anyone using a universal remote to control their Roku TV.

### PR DESCRIPTION
**Pull request by ApexArray**
**Changes**
- implement play/pause action on ok button press
- prevent handling any buttons while trick play bar is visible (otherwise, ok button would play/pause video in the background instead of making a selection during seeking)

**Issues**
Fixes #883

### CODE REVIEW

**Comment by ApexArray** [comment]
Note this will play/pause the video but will not display the trick play bar. I couldn't find an easy way to do that reliably without implementing a completely custom trick play bar.

https://community.roku.com/t5/Roku-Developer-Program/no-trickPlayBar-after-pausing-video-programmatically/m-p/470009#M37860
https://community.roku.com/t5/Roku-Developer-Program/SceneGraph-video-playback-positioning/m-p/475056

One possible workaround is sending `POST /press/Play` to the Roku's own API, which should fully emulate the play/pause button. Would that be considered too much of a hack?
https://community.roku.com/t5/Roku-Developer-Program/Is-localhost-127-0-0-1-supported/m-p/319043

File: components/JFVideo.brs [hunk 1]
```diff
@@ -222,7 +222,12 @@ end sub
 
 function onKeyEvent(key as string, press as boolean) as boolean
 
-    if key = "OK" and m.nextEpisodeButton.hasfocus() and not m.top.trickPlayBar.visible
+    ' don't do anything if trick play bar is visible
+    if m.top.trickPlayBar.visible
```
Comment (1hitsong) [inline components/JFVideo.brs]: This check changes existing functionality. Previously I could bring up the video info or the subtitle selection while the video is paused. The addition of this check prevents that.

Perhaps we should move this condition into the new "OK" check.

File: components/JFVideo.brs [hunk 2]
```diff
@@ -243,6 +248,16 @@ function onKeyEvent(key as string, press as boolean) as boolean
     else if key = "up"
         m.top.selectPlaybackInfoPressed = true
         return true
+    else if key = "OK"
+        print("OK pressed")
```
Comment (1hitsong) [inline components/JFVideo.brs]: Debug code

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Review by ApexArray — COMMENTED** [review]
done

**Review by 1hitsong — APPROVED** [review]

**Comment by 1hitsong** [comment]
To be merged 12/22 unless issue arises.

**Comment by ApexArray** [comment]
I found another issue with this. It's fairly minor, but it's definitely not expected behavior.
- Pause the video with the play button
- Trickplay bar becomes visible
- The OK button cannot resume the video until the Trickplay bar fades out

We're using the trickplay visibility condition to avoid overriding the default OK button behavior during seeking. But, when not seeking, the OK button should pause/resume as expected.

Possible solutions:
1. Temporarily disable the OK button functionality when left/right button is pressed.
2. Emulate the play button by sending `POST /press/Play` to the Roku API.
3. Find a way to detect a seeking state (I could not find a way to do this)

We could also choose to leave this behavior unhandled, but I don't really like that idea.

**Comment by ApexArray** [comment]
Actually, I found a more subtle fix. We can remove the trickplay check all together and return false to tell Roku the keypress was not handled. This allows the button to continue performing its default behavior during seeking.

My only concern is this paragraph from the developer documentation.

>The onKeyEvent() function must return true if the XML component handled the event, or false if it did not handle the event. Returning false allows the event to continue bubbling up the focus chain so that ancestors of the XML component can handle the event. 

We're technically handling the event, but we _want_ it to continue bubbling up the focus chain. Would this cause an issue with channel certification?

**Comment by neilsb** [comment]
Pretty sure that won't cause a problem with certification, and from a software design point of view it's a totally legitimate implementation. 

Many things can look look for an event and even act on it, but not "handle" it. Handling it a decision that's being made saying "I've done all that's required and nothing more should happen due to this even being raised".

**Pull request merged**